Assets/Project/Modules/VFX/Generic/Scripts/ParticleBehaviours/MaterialInterpolatorRecycleParticle.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using Cysharp.Threading.Tasks;
+using DG.Tweening;
+using Popeye.Core.Pool;
+using UnityEngine;
+using UnityEngine.Serialization;
+
+namespace Popeye.Modules.VFX.Generic.ParticleBehaviours
+{
+    public class MaterialInterpolatorRecycleParticle : RecyclableObject
+    {
+        [SerializeField] private MeshRenderer _meshRenderer;
+        private Material _material;
+
+        [System.Serializable]
+        private class FloatSetupData
+        {
+            public string id;
+            public float initialValue;
+        }
+        
+        [System.Serializable]
+        private class FloatInterpolationData
+        {
+            public float delay;
+            
+            public string id;
+            public float duration;
+            public float endValue = 1.0f;
+            public Ease ease;
+            
+            public bool waitForCompletion;
+        }
+
+        [SerializeField] private List<FloatSetupData> _floatSetupDatas = new();
+        [SerializeField] private List<FloatInterpolationData> _floatInterpolationDatas = new();
+
+        private void Awake()
+        {
+            _material = _meshRenderer.material;
+        }
+
+        internal override void Init()
+        {
+            Setup();
+            FloatInterpolations();
+        }
+
+        private void Setup()
+        {
+            foreach (var data in _floatSetupDatas)
+            {
+                _material.SetFloat(data.id, data.initialValue);
+            }
+        }
+        
+        private async UniTaskVoid FloatInterpolations()
```

Comment:
This is taking it one step further, but we could abstract this entire function into a new class "MaterialInterpolator",
with an `async UniTask ApplyInterpolations(Material, MaterialInterpolationData[])`   which we can pass the material and list of MaterialInterpolationData.

It seems unnecessary, but I imagine you would want to apply sequence of interpolations in other places outside  the MaterialInterpolatorRecycleParticle class. If we do this we abstract the code and avoid repeating it.

Reply:
If you need to, you can pass the Transform too (if you want to modify scale or rotation). 
Though I encourage you to try make it all work in the shader, if possible interpolating only one "t" parameter, in which case there wouldn't be much need to create this "MaterialInterpolator" class, still, I think it would be nice regardless.
Anyways, in the end do as you think is best.